extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapObserver.kt

```diff
@@ -0,0 +1,119 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.SurfaceCallback
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.ScreenCoordinate
+
+/**
+ * Many downstream services will not work until the surface has been created and the map has
+ * loaded. This interface allows you to create custom Mapbox experiences for the car.
+ */
+interface MapboxCarMapObserver {
+
+  /**
+   * Called when a [MapboxCarMapSurface] has been loaded.
+   * You can assume there will only be a single surface at a time.
+   *
+   * @see [MapboxCarMap.registerObserver]
+   *
+   * @param mapboxCarMapSurface loaded and ready to use car map surface
+   */
+  fun loaded(mapboxCarMapSurface: MapboxCarMapSurface) {
+    // No op by default
+  }
+
+  /**
+   * Called when the car library updates the visible regions for the surface. For example, this
+   * is triggered when the action buttons come in and out of visibility.
+   * You can assume this will be called after [loaded].
+   *
+   * @param visibleArea the visible area provided by the host
+   * @param edgeInsets distance from each side of the screen that creates the [visibleArea]
+   */
+  fun visibleAreaChanged(visibleArea: Rect, edgeInsets: EdgeInsets) {
+    // No op by default
+  }
+
+  /**
+   * Allows you to implement or observe the map scroll gesture handler. The surface is [loaded]
+   * before this can be triggered.
+   *
+   * @see [SurfaceCallback.onScroll] for instructions to enable.
+   *
+   * @param mapboxCarMapSurface loaded and ready car map surface
+   * @param distanceX the distance in pixels along the X axis
+   * @param distanceY the distance in pixels along the Y axis
+   *
+   * @return true when the fling scroll was handled, false will trigger the default handler
+   */
+  fun scroll(
```

Comment:
nit : align namings? Either `scrolled`, `flinged` or `visibleAreaChange` / `load` above. 

Maybe even consider `onScroll`, `onFling`, `onLoad` ?

Reply:
These are action events. The implementation is handling the events. For example, if you have a `CarCameraController` that implements this interface. It will need to update the camera position when scroll is called.

Agree that this could be odd though, that an "Observer" is handling events. Maybe `MapboxCarMapObserver` should be called `MapboxCarMapInteractor`

Reply:
I considered breaking this into 2 separate interfaces. Ended up going with 1 interface because it is simpler interactions. Also have spent some time considering different names, but ended up with what is here.

- [x] MapboxCarMapObserver
- [ ] MapboxCarMapCallback
- [ ] MapboxSurfaceCallback
- [ ] MapboxCarMapInteractor
...

Reply:
Ah, ok, if semantics is different - let's keep it as is

Reply:
@yunikkk sorry i'm going back and forth here. also just noticed your "lets keep it comment". I pushed the taking the recommendation and pushed the "Maybe even consider `onScroll`, `onFling`, `onLoad`" 😄 

could be better because it is also more consistent with `SurfaceCallback` and other map functions.

Although, instead of "onLoad" changed it to be "onAttached" to accompany "onDetached"

Reply:
Sounds good, too!